### PR TITLE
AccessControl: Make `Permissions` component customizable

### PR DESCRIPTION
**Pull request by gamab**
**What this PR does / why we need it**:
In the case of teams, having the title set to `Permissions` is a bit misleading since it doesn't match the "`Members`" tab name.

In this PR we intend to make the component customizable and slightly more generic.

Currently for datasources, the view looks like this:
Capture d’écran de 2022-01-19 11-38-21

This PR sets titles to the tables instead:
Capture d’écran de 2022-01-19 11-37-21

And could **theoretically** be configured as follow:
image

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/AccessControl/Permissions.tsx [hunk 1]
```diff
@@ -20,14 +20,15 @@ const INITIAL_DESCRIPTION: Description = {
...
+  title?: string;
   resource: string;
   resourceId: number;
 
   canListUsers: boolean;
   canSetPermissions: boolean;
 };
 
-export const Permissions = ({ resource, resourceId, canListUsers, canSetPermissions }: Props) => {
+export const Permissions = ({ title, resource, resourceId, canListUsers, canSetPermissions }: Props) => {
```
Comment (Clarity-89) [inline public/app/core/components/AccessControl/Permissions.tsx:31]:
```suggestion
export const Permissions = ({ resource, resourceId, canListUsers, canSetPermissions, title='Permissions' }: Props) => {
```
You can set a default title instead, which will be used if no `title` prop is provided.

File: public/app/core/components/AccessControl/Permissions.tsx [hunk 2]
```diff
@@ -114,7 +115,7 @@ export const Permissions = ({ resource, resourceId, canListUsers, canSetPermissi
   return (
     <div>
       <div className="page-action-bar">
-        <h3 className="page-sub-heading">Permissions</h3>
+        <h3 className="page-sub-heading">{title ? title : 'Permissions'}</h3>
```
Comment (Clarity-89) [inline public/app/core/components/AccessControl/Permissions.tsx:118]:
```suggestion
        <h3 className="page-sub-heading">{title}</h3>
```

File: public/app/core/components/AccessControl/PermissionList.tsx [hunk 3]
```diff
@@ -3,23 +3,31 @@ import { ResourcePermission } from './types';
 import { PermissionListItem } from './PermissionListItem';
 
 interface Props {
-  title: string;
+  type: string;
```
Comment (Clarity-89) [inline public/app/core/components/AccessControl/PermissionList.tsx:6]: `type` is an overloaded term and I think it's quite confusing here, doesn't `title` work in this case? it seems like this prop is used as a title.

**Review by Clarity-89 — APPROVED** [review]
Looks good, just one comment.

File: public/app/core/components/AccessControl/PermissionList.tsx (hunk 3 above)
Comment (gamab) [reply public/app/core/components/AccessControl/PermissionList.tsx:6]: Thanks I wasn't sure this change was for the best anyway, so i'll revert on that !

**Pull request merged**